### PR TITLE
Refactor/update simple types

### PR DESCRIPTION
**Pull request by aviveiros11**
See proposal.

### CODE REVIEW

**Review by macintoshpie — COMMENTED** [review]
Nice! I think this is great, will hopefully help avoid unexpected issues for folks down the road.

I think we still need to:
1. get @Ryoken 's recent changes from #349 merged into v3
2. set the merge target of this PR to v3 and resolve conflicts in this branch
3. merrrge

File: BuildingSync.xsd [hunk 1]
```diff
@@ -709,7 +709,7 @@
         </xs:annotation>
         <xs:complexType>
           <xs:simpleContent>
-            <xs:extension base="xs:integer">
+            <xs:extension base="xs:nonNegativeInteger">
```
Comment (JieXiong9119) [inline BuildingSync.xsd:712]: Why is `Perimeter` limited to `integer` but not 'decimal`? @nllong

File: BuildingSync.xsd [hunk 2]
```diff
@@ -1794,7 +1794,7 @@
...
```
Comment (JieXiong9119) [inline BuildingSync.xsd:1797]: Again, why is `ConditionedVolume` limited to `integer`?

**Review by JieXiong9119 — APPROVED** [review]
`nonNegativeInteger` looks great!
Got two minor questions on the general 'integer/decimal' restriction on `Perimeter` and `ConditionedVolume`.

**Pull request merged**